packages/bundler-plugin-core/src/index.ts

```diff
@@ -106,7 +111,7 @@ export function sentryUnpluginFactory({
 
     const options = normalizeUserOptions(userOptions);
 
-    if (unpluginMetaContext.watchMode || options.disable) {
```

Comment:
I know this is technically not the goal of this PR but thinking about this logic more holistically we probably don't want to disable the entire plugin for dev mode because we have other things that may be valuable in dev mode, like react component annotations or tree shaking utilities (I guess for debugging purposes).

Should we limit this check to just anything that has a side-effect/is invoking Sentry CLI?